google-cloud-bigquerystorage/src/main/java/com/google/cloud/bigquery/storage/v1alpha2/StreamWriter.java

```diff
@@ -0,0 +1,856 @@
+            .setEndpoint(builder.endpoint)
+            .build();
+    shutdown = new AtomicBoolean(false);
+    refreshAppend();
+  }
+
+  /** Stream name we are writing to. */
+  public String getStreamNameString() {
+    return streamName;
+  }
+
+  /**
+   * Schedules the writing of a message. The write of the message may occur immediately or be
+   * delayed based on the writer batching options.
+   *
+   * <p>Example of writing a message.
+   *
+   * <pre>{@code
+   * AppendRowsRequest message;
+   * ApiFuture<AppendRowsResponse> messageIdFuture = writer.append(message);
+   * ApiFutures.addCallback(messageIdFuture, new ApiFutureCallback<AppendRowsResponse>() {
+   *   public void onSuccess(AppendRowsResponse response) {
+   *     if (response.hasOffset()) {
+   *       System.out.println("written with offset: " + response.getOffset());
+   *     } else {
+   *       System.out.println("received an in stream error: " + response.error().toString());
+   *     }
+   *   }
+   *
+   *   public void onFailure(Throwable t) {
+   *     System.out.println("failed to write: " + t);
+   *   }
+   * }, MoreExecutors.directExecutor());
+   * }</pre>
+   *
+   * @param message the message in serialized format to write to BigQuery.
+   * @return the message ID wrapped in a future.
+   */
+  public ApiFuture<AppendRowsResponse> append(AppendRowsRequest message) {
+    Preconditions.checkState(!shutdown.get(), "Cannot append on a shut-down writer.");
+
+    final AppendRequestAndFutureResponse outstandingAppend =
+        new AppendRequestAndFutureResponse(message);
+    List<InflightBatch> batchesToSend;
+    messagesBatchLock.lock();
+    try {
+      batchesToSend = messagesBatch.add(outstandingAppend);
+      // Setup the next duration based delivery alarm if there are messages batched.
+      setupAlarm();
+      if (!batchesToSend.isEmpty()) {
+        for (final InflightBatch batch : batchesToSend) {
+          LOG.fine("Scheduling a batch for immediate sending.");
+          writeBatch(batch);
+        }
+      }
+    } finally {
+      messagesBatchLock.unlock();
+    }
+
+    return outstandingAppend.appendResult;
+  }
+
+  public void refreshAppend() throws IOException {
+    synchronized (this) {
+      Preconditions.checkState(!shutdown.get(), "Cannot append on a shut-down writer.");
+      if (stub != null) {
+        stub.shutdown();
+      }
+      backgroundResourceList.remove(stub);
+      stub = BigQueryWriteClient.create(stubSettings);
+      backgroundResourceList.add(stub);
+      backgroundResources = new BackgroundResourceAggregation(backgroundResourceList);
+      messagesBatch.resetAttachSchema();
+      bidiStreamingCallable = stub.appendRowsCallable();
+      clientStream = bidiStreamingCallable.splitCall(responseObserver);
+      try {
+        while (!clientStream.isSendReady()) {
+          Thread.sleep(100);
+        }
+      } catch (InterruptedException e) {
```

Comment:
is there a reason why we're ignoring this exception? see: https://google.github.io/styleguide/javaguide.html#s6.2-caught-exceptions

Reply:
This is to wait for channel to be ready. It is a common practice to wait on such things. If there are better guidelines on how to use the streaming API, please let me know.

Reply:
I guess I was asking why we're catching an exception but not doing anything with it? see https://google.github.io/styleguide/javaguide.html#s6.2-caught-exceptions for best practice on exception handling -- if it's expected, maybe use `InterruptedException expected` instead of just `InterruptedException e` to be more descriptive?

Reply:
Done.